lib/Parse/ParseDecl.cpp

```diff
@@ -3703,14 +3728,18 @@ void Parser::ParseStructUnionBody(SourceLocation RecordLoc,
     }
 
     if (!Tok.is(tok::at)) {
-      auto CFieldCallback = [&](ParsingFieldDeclarator &FD) {
+      auto CFieldCallback = [&](ParsingFieldDeclarator &FD,
+                                CachedTokens *BoundsExprTokens) {
         // Install the declarator into the current TagDecl.
         Decl *Field =
             Actions.ActOnField(getCurScope(), TagDecl,
                                FD.D.getDeclSpec().getSourceRange().getBegin(),
                                FD.D, FD.BitfieldSize);
         FieldDecls.push_back(Field);
         FD.complete(Field);
+        if (BoundsExprTokens != nullptr) {
+          deferredBoundsExpressions.emplace_back(Field, BoundsExprTokens);
```

Comment:
I'll update the code to use unique_ptr's instead, removing the ambiguity about when things will be freed.